src/wp-admin/includes/class-wp-filesystem-ssh2.php

```diff
@@ -150,7 +150,8 @@ public function connect() {
 				return false;
 			}
 		} else {
-			if ( ! @ssh2_auth_pubkey_file( $this->link, $this->options['username'], $this->options['public_key'], $this->options['private_key'], $this->options['password'] ) ) {
+			$password = isset( $this->options['password'] ) ? $this->options['password'] : null;
```

Comment:
On second thought, we can just set `$this->options['password'] = null;` in an `else` block [here](https://github.com/WordPress/wordpress-develop/blob/e61d17a5ed156c57104f214846c25224131f49a4/src/wp-admin/includes/class-wp-filesystem-ssh2.php#L104-L106).

So:
```php
if ( ! $this->keys ) {
	$this->errors->add( 'empty_password', __( 'SSH2 password is required' ) );
} else {
	$this->options['password'] = null;
}
```
@J-Dill Are you happy to make this change to the PR?

Reply:
Yes that seems like a reasonable way to do it as well. I will make that change here shortly.

Reply:
@costdev Done. Please take another look.

Reply:
Thanks! That looks good to me!